### ISSUE TITLE
Fix S1144 FP: using an explicit record constructor from a positional subtype

### ISSUE THREAD

**Issue by shuebner** [SonarSource/sonar-dotnet#5381]
### Description

When a private constructor of a record is only used by a nested subclass using positional syntax, S1144 will report the private constructor as unused, when in fact it is not.

### Repro steps

```csharp
namespace S1144Repro
{
    public abstract record Foo
    {
        Foo(string value) // will trigger S1144
        {
            Value = value;
        }

        public string Value { get; }

        public sealed record Bar(string Value) : Foo(Value);
    }
}
```

### Expected behavior

S1144 does not trigger.

### Actual behavior

S1144 triggers.
![image](https://user-images.githubusercontent.com/1770684/153543243-3bbf24f7-8405-4d3c-92ed-23a0b68255d5.png)

### Known workarounds

Declaring the constructor as `private protected` avoids the FP, but changes semantics.

### Related information

* C#/VB.NET Plugins version: 8.35.0.42613
* Visual Studio version: 16.11.10
* MSBuild / dotnet version: 5.0.405
* SonarScanner for .NET version (if used): ---
* Operating System: Windows 11

**Comment by shuebner** [comment]
bump
